src/kibana/plugins/discover/index.html

```diff
@@ -148,17 +149,30 @@
           <!-- result -->
           <div class="results" ng-show="resultState === 'ready'">
             <div class="discover-timechart" ng-if="opts.timefield">
-              <center class="small">
-                <span tooltip="To change the time, click the clock icon in the navigation bar">{{timeRange.from | moment}} - {{timeRange.to | moment}}</span>
-                <!-- TODO: Currently no way to apply this setting to the visualization -->
-                <!-- <select
-                  class="form-control"
-                  ng-model="state.interval"
-                  ng-options="interval as interval for interval in intervalOptions"
-                  ng-change="setupVisualization();fetch()"
-                  >
-                </select> -->
-              </center>
+              <header>
+                <center class="small">
+                  <span tooltip="To change the time, click the clock icon in the navigation bar">{{timeRange.from | moment}} - {{timeRange.to | moment}}</span>
+
+                  &mdash;
+
+                  <span class="results-interval" ng-hide="showInterval">
+                    <a
+                      ng-click="$parent.showInterval = !$parent.showInterval">
```

Comment:
Accessing $parent to set the value from the view seems wrong. Maybe change this to be a function on the parent that sets the value. Eg:

```
    $scope.toggleInterval = function () {
      $scope.showInterval =  !$scope.showInterval;
    };
```